src/test/java/core/basesyntax/dao/FruitDaoImplTest.java

```diff
@@ -0,0 +1,142 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitDaoImplTest {
+    private static FruitDao fruitDao;
+    private static FruitTransaction fruitTransaction;
+    private static Exception exception;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitDao = new FruitDaoImpl();
+        fruitTransaction = new FruitTransaction();
+    }
+
+    @Before
+    public void setUp() {
+        fruitTransaction.setFruit("apple");
+        fruitTransaction.setQuantity(10);
+        fruitTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        exception = new Exception();
+    }
+
+    @Test
+    public void add_nullFruitTransaction_notOK() {
+        try {
+            fruitDao.add(null);
+        } catch (Exception e) {
+            exception = e;
+        }
+        Assert.assertSame(RuntimeException.class, exception.getClass());
+        Assert.assertEquals(0, Storage.fruits.size());
+    }
+
+    @Test
+    public void add_nullFruitName_notOK() {
+        fruitTransaction.setFruit(null);
+        try {
+            fruitDao.add(fruitTransaction);
+        } catch (Exception e) {
+            exception = e;
+        }
+        Assert.assertSame(RuntimeException.class, exception.getClass());
```

Comment:
```suggestion
           Assert.assertSame(RuntimeException.class, e.getClass());
        }
```
the same for other similar cases